### PR TITLE
Graphite support

### PR DESCRIPTION
**Pull request by alfredodeza**

### CODE REVIEW

File: ansible/roles/graphite/tasks/carbon.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    state: present
+    backrefs: true
+  sudo: true
+
+- name: create the rewrite config with the secret api key
+  template:
+    src: ../templates/rewrite-rules.conf.j2
+    dest: "/etc/carbon/rewrite-rules.conf"
+  notify:
+    - restart carbon
```
Comment (andrewschoen) [inline ansible/roles/graphite/tasks/carbon.yml]: I don't see this hander in ``graphite/handlers/main.yml``

File: ansible/roles/graphite/tasks/postgresql.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    src: ../templates/local_settings.py.j2
+    dest: "/etc/graphite/local_settings.py"
+  notify:
+    - restart app
+  sudo: true
+
+  # there is a bug where if you don't migrate auth first only it will fail
+  # with "ProgrammingError: relation "auth_user" does not exist"
+- name: run migrate for auth first
+  command: graphite-manage migrate --noinput auth
```
Comment (andrewschoen) [inline ansible/roles/graphite/tasks/postgresql.yml]: Does this task need ``sudo: true`` as well? It looks like you've done that below with the other ``migrate`` command.

File: ansible/roles/graphite/tasks/postgresql.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+- name: run syncdb to ensure database schema
```
Comment (andrewschoen) [inline ansible/roles/graphite/tasks/postgresql.yml]: nit: you're not running ``syncdb`` here.

File: ansible/roles/graphite/templates/local_settings.py.j2 [hunk 4]
```diff
@@ -0,0 +1,222 @@
+## Graphite local_settings.py
```
Comment (andrewschoen) [inline ansible/roles/graphite/templates/local_settings.py.j2]: Do we want an ``ansible_managed`` at the top of this template?

File: ansible/roles/graphite/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    name: {{ item }}
+    state: present
+  with_items: system_packages
+  tags:
+    - packages
+
+- name: copy graphite.wsgi to graphite-web location
+  command:
+    creates: "/usr/lib/python2.7/dist-packages/graphite/graphite_web.py"
+    cp /usr/share/graphite-web/graphite.wsgi /usr/lib/python2.7/dist-packages/graphite/graphite_web.py
```
Comment (andrewschoen) [inline ansible/roles/graphite/tasks/main.yml]: This doesn't look right, shouldn't it be on the same line as ``command:``?

File: ansible/roles/graphite/tasks/systemd.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+---
+
+- name: ensure /var/log/graphite dir exists
+  sudo: true
+  file: path=/var/log/graphite state=directory owner=_graphite group=_graphite recurse=yes
```
Comment (andrewschoen) [inline ansible/roles/graphite/tasks/systemd.yml]: Can we use the ``key: value`` each option on it's own line format in this file?

File: ansible/roles/graphite/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+- include: postgresql.yml
+  tags:
+    - postgresql
+
+- name: ensure graphite is running
+  sudo: true
+  service:
+    name: graphite
+    state: restarted
```
Comment (andrewschoen) [inline ansible/roles/graphite/tasks/main.yml]: Do you want ``started`` here instead of ``restarted``?  Maybe it doesn't matter?

Comment (alfredodeza) [inline ansible/roles/graphite/tasks/main.yml]: I think `restarted` is what we want (need) here.

**Comment by andrewschoen** [comment]
test this please

**Pull request merged**